diff --git a/src/main/java/org/broad/igv/track/TrackLoader.java b/src/main/java/org/broad/igv/track/TrackLoader.java
index f09591e39..f87d019b6 100644
--- a/src/main/java/org/broad/igv/track/TrackLoader.java
+++ b/src/main/java/org/broad/igv/track/TrackLoader.java
@@ -76,6 +76,8 @@ import org.broad.igv.renderer.MutationRenderer;
 import org.broad.igv.renderer.PointsRenderer;
 import org.broad.igv.sam.*;
 import org.broad.igv.sam.reader.IndexNotFoundException;
+import org.broad.igv.scanfold.ScanfoldlessDBNLoader;
+import org.broad.igv.scanfold.VarnaLoader;
 import org.broad.igv.tdf.TDFDataSource;
 import org.broad.igv.tdf.TDFReader;
 import org.broad.igv.ui.IGV;
@@ -207,6 +209,10 @@ public class TrackLoader {
                 loadMutFile(locator, newTracks, genome); // Must be tried before ".maf" test below
             } else if (format.equals("maf")) {
                 loadMultipleAlignmentTrack(locator, newTracks, genome);
+            } else if (format.equals("scanfoldlessdbn")) {
+                ScanfoldlessDBNLoader.loadFofnTwo(locator, newTracks, genome);
+            } else if (format.equals("scanfoldvarna")) {
+                VarnaLoader.loadFofn(locator, genome);
             } else {
                 //if a url, try htsget
                 boolean isHtsget = tryHtsget(locator, newTracks, genome);
diff --git a/src/main/java/org/broad/igv/ui/IGVMenuBar.java b/src/main/java/org/broad/igv/ui/IGVMenuBar.java
index 4d46a5a26..2c7ec796d 100644
--- a/src/main/java/org/broad/igv/ui/IGVMenuBar.java
+++ b/src/main/java/org/broad/igv/ui/IGVMenuBar.java
@@ -62,6 +62,10 @@ import org.broad.igv.util.HttpUtils;
 import org.broad.igv.util.LongRunningTask;
 import org.broad.igv.util.blat.BlatClient;
 import org.broad.igv.util.encode.EncodeFileBrowser;
+import org.broad.igv.feature.Strand;
+import org.broad.igv.scanfold.RNAFoldGui;
+import org.broad.igv.scanfold.RNAStructureGui;
+import org.broad.igv.scanfold.ScanFoldGui;
 
 import javax.swing.*;
 import javax.swing.event.MenuEvent;
@@ -173,6 +177,7 @@ public class IGVMenuBar extends JMenuBar implements IGVEventObserver {
         menus.add(createViewMenu());
         menus.add(createTracksMenu());
         menus.add(createRegionsMenu());
+        menus.add(createScanfoldMenu());
 
         refreshToolsMenu();
         menus.add(toolsMenu);
@@ -437,7 +442,7 @@ public class IGVMenuBar extends JMenuBar implements IGVEventObserver {
         menuAction.setToolTipText("Load a FASTA, .json, or .genome file...");
         menuItems.add(MenuAndToolbarUtils.createMenuItem(menuAction));
 
-        // Download genome from server 
+        // Download genome from server
         menuAction = new MenuAction("Select Hosted Genome...", null) {
             @Override
             public void actionPerformed(ActionEvent event) {
@@ -1221,4 +1226,46 @@ public class IGVMenuBar extends JMenuBar implements IGVEventObserver {
 
         return menuItem;
     }
+
+    private JMenu createScanfoldMenu() {
+        List<JComponent> menuItems = new ArrayList<JComponent>();
+        MenuAction extrasMenuAction = new MenuAction("ScanFold");
+        MenuAction scanfoldMenuAction = new MenuAction("Run ScanFold on visible region...", null, 0) {
+            @Override
+            public void actionPerformed(ActionEvent e) {
+            	final ReferenceFrame defaultFrame = FrameManager.getDefaultFrame();
+                String chr = defaultFrame.getChrName();
+                int start = (int) defaultFrame.getOrigin();
+                int end = (int) defaultFrame.getEnd();
+                String sequence = ScanFoldGui.extractSequence(GenomeManager.getInstance().getCurrentGenome(), chr, start, end, Strand.NONE);
+                ScanFoldGui.launch(chr, start, sequence, false, "main");
+            }
+        };
+        menuItems.add(MenuAndToolbarUtils.createMenuItem(scanfoldMenuAction));
+        MenuAction rnafoldMenuAction = new MenuAction("Run RNAFold on visible region...", null, 0) {
+            @Override
+            public void actionPerformed(ActionEvent e) {
+            	final ReferenceFrame defaultFrame = FrameManager.getDefaultFrame();
+                String chr = defaultFrame.getChrName();
+                int start = (int) defaultFrame.getOrigin();
+                int end = (int) defaultFrame.getEnd();
+                String sequence = RNAFoldGui.extractSequence(GenomeManager.getInstance().getCurrentGenome(), chr, start, end, Strand.NONE);
+                RNAFoldGui.launch(chr, start, sequence, false, "main");
+            }
+        };
+        menuItems.add(MenuAndToolbarUtils.createMenuItem(rnafoldMenuAction));
+        MenuAction rnastructureMenuAction = new MenuAction("Run RNAStructure on visible region...", null, 0) {
+            @Override
+            public void actionPerformed(ActionEvent e) {
+            	final ReferenceFrame defaultFrame = FrameManager.getDefaultFrame();
+                String chr = defaultFrame.getChrName();
+                int start = (int) defaultFrame.getOrigin();
+                int end = (int) defaultFrame.getEnd();
+                String sequence = RNAStructureGui.extractSequence(GenomeManager.getInstance().getCurrentGenome(), chr, start, end, Strand.NONE);
+                RNAStructureGui.launch(chr, start, sequence, false, "main");
+            }
+        };
+        menuItems.add(MenuAndToolbarUtils.createMenuItem(rnastructureMenuAction));
+        return MenuAndToolbarUtils.createMenu(menuItems, extrasMenuAction);
+    }
 }
diff --git a/src/main/java/org/broad/igv/ui/panel/RegionOfInterestPanel.java b/src/main/java/org/broad/igv/ui/panel/RegionOfInterestPanel.java
index c329652b7..f85131be2 100644
--- a/src/main/java/org/broad/igv/ui/panel/RegionOfInterestPanel.java
+++ b/src/main/java/org/broad/igv/ui/panel/RegionOfInterestPanel.java
@@ -35,6 +35,9 @@ import org.broad.igv.util.blat.BlatClient;
 import org.broad.igv.feature.RegionOfInterest;
 import org.broad.igv.feature.genome.Genome;
 import org.broad.igv.feature.genome.GenomeManager;
+import org.broad.igv.scanfold.RNAFoldGui;
+import org.broad.igv.scanfold.RNAStructureGui;
+import org.broad.igv.scanfold.ScanFoldGui;
 import org.broad.igv.ui.IGV;
 import org.broad.igv.util.LongRunningTask;
 import org.broad.igv.util.NamedRunnable;
@@ -204,7 +207,39 @@ public class RegionOfInterestPanel extends JPanel {
         }
         popupMenu.add(item);
 
+        item = new JMenuItem("Run ScanFold on selected region...");
+        item.addActionListener(new ActionListener() {
 
+            public void actionPerformed(ActionEvent e) {
+            	Genome genome = GenomeManager.getInstance().getCurrentGenome();
+            	String sequence = ScanFoldGui.extractSequence(genome, roi.getChr(), roi.getStart(), roi.getEnd(), Strand.NONE);
+            	ScanFoldGui.launch(roi.getChr(), roi.getStart(), sequence, false, "roi");
+            }
+        });
+        popupMenu.add(item);
+        
+        item = new JMenuItem("Run RNAFold on selected region...");
+        item.addActionListener(new ActionListener() {
+
+            public void actionPerformed(ActionEvent e) {
+            	Genome genome = GenomeManager.getInstance().getCurrentGenome();
+            	String sequence = RNAFoldGui.extractSequence(genome, roi.getChr(), roi.getStart(), roi.getEnd(), Strand.NONE);
+            	RNAFoldGui.launch(roi.getChr(), roi.getStart(), sequence, false, "roi");
+            }
+        });
+        popupMenu.add(item);
+        
+        item = new JMenuItem("Run RNAStructure on selected region...");
+        item.addActionListener(new ActionListener() {
+
+            public void actionPerformed(ActionEvent e) {
+            	Genome genome = GenomeManager.getInstance().getCurrentGenome();
+            	String sequence = RNAStructureGui.extractSequence(genome, roi.getChr(), roi.getStart(), roi.getEnd(), Strand.NONE);
+            	RNAStructureGui.launch(roi.getChr(), roi.getStart(), sequence, false, "roi");
+            }
+        });
+        popupMenu.add(item);
+        
         popupMenu.add(new JSeparator());
 
         item = new JMenuItem("Delete");
